src/alloc.c

```diff
@@ -308,6 +308,22 @@ DLLEXPORT jl_value_t *jl_new_struct_uninit(jl_datatype_t *type)
     return jv;
 }
 
+DLLEXPORT jl_bytevec_struct_t jl_bytevec(const uint8_t *data, size_t n)
```

Comment:
It's not returning a `jl_value_t` – `jl_bytevec_struct_t` is a C struct not a Julia value. The `ByteVec` type happens to have as its data a value that this struct is well-suited to manipulating.